climada/hazard/test/test_landslide.py

```diff
@@ -19,132 +19,65 @@
-        empty_LS.set_raster(
-            [DATA_DIR.joinpath('test_global_landslide_nowcast_20190501.tif')],
-            window=Window(window_array[0], window_array[1], window_array[3], window_array[2]))
-        empty_LS._intensity_cat_to_prob(max_prob=0.0001)
-        self.assertTrue(max(empty_LS.intensity_cat.data) == 2)
-        self.assertTrue(min(empty_LS.intensity_cat.data) == 1)
-        self.assertTrue(max(empty_LS.intensity.data) == 0.0001)
-        self.assertTrue(min(empty_LS.intensity.data) == 0)
-
-    def test_intensity_prob_to_binom(self):
-        empty_LS = Landslide()
-        window_array = empty_LS._get_window_from_coords(
-            path_sourcefile=DATA_DIR.joinpath('test_global_landslide_nowcast_20190501.tif'),
-            bbox=[47, 23, 46, 22])
-        empty_LS.set_raster(
-            [DATA_DIR.joinpath('test_global_landslide_nowcast_20190501.tif')],
-            window=Window(window_array[0], window_array[1], window_array[3], window_array[2]))
-        empty_LS._intensity_cat_to_prob(max_prob=0.0001)
-        empty_LS._intensity_prob_to_binom(100)
-        self.assertTrue(max(empty_LS.intensity_prob.data) == 0.0001)
-        self.assertTrue(min(empty_LS.intensity_prob.data) == 0)
-        self.assertTrue(max(empty_LS.intensity.data) == 1)
-        self.assertTrue(min(empty_LS.intensity.data) == 0)
-
-    def test_intensity_binom_to_range(self):
-        empty_LS = Landslide()
-        window_array = empty_LS._get_window_from_coords(
-            path_sourcefile=DATA_DIR.joinpath('test_global_landslide_nowcast_20190501.tif'),
-            bbox=[47, 23, 46, 22])
-        empty_LS.set_raster(
-            [DATA_DIR.joinpath('test_global_landslide_nowcast_20190501.tif')],
-            window=Window(window_array[0], window_array[1], window_array[3], window_array[2]))
-        empty_LS._intensity_cat_to_prob(max_prob=0.0001)
-        empty_LS._intensity_prob_to_binom(100)
-        empty_LS.check()
-        empty_LS.centroids.set_meta_to_lat_lon()
-        empty_LS.centroids.set_geometry_points()
-        empty_LS._intensity_binom_to_range(max_dist=1000)
-        self.assertTrue(
-            len(empty_LS.intensity.data[(empty_LS.intensity.data > 0)
-                                        & (empty_LS.intensity.data < 1)]) > 0)
-
-    def test_get_hist_events(self):
-        empty_LS = Landslide()
-        bbox = [48, 23, 40, 20]
-        COOLR_path = DATA_DIR.joinpath('nasa_global_landslide_catalog_point.shp')
-        LS_catalogue_part = empty_LS._get_hist_events(bbox, COOLR_path)
-        self.assertTrue(max(LS_catalogue_part.latitude) <= bbox[0])
-        self.assertTrue(min(LS_catalogue_part.latitude) >= bbox[2])
-        self.assertTrue(max(LS_catalogue_part.longitude) <= bbox[1])
-        self.assertTrue(min(LS_catalogue_part.longitude) >= bbox[3])
-
-
+LS_HIST_FILE = DATA_DIR / 'test_ls_hist.shp'
+LS_PROB_FILE = DATA_DIR / 'test_ls_prob.tif'
+
+class TestLandslideModule(unittest.TestCase):     
```

Comment:
Please add a test for a country that crosses the Anti-meridian (e.g. the Fidji or the US). I am not sure that the bounding boxes in the code can deal with this.

Reply:
True that. it does not.
But in my opinion, this is nothing that has to do with the landslide module, but with the read_raster function, which does the reading of raster files incl bounding boxes of shapely (box) geometries and where this originates from.

I will open an issue on that. We can write sth where, if it a bbox crosses the anti-meridian, it actually reads in two files (left & right of the anti-meridian) and combines them

Reply:
alternatively, i could also just write a wrapper in my landslide module that does that, but i dont find this very clean, since it applies to _any_ raster files that are read in from a bbox

Reply:
I agree with your point. It does not make sense to have a special solutions just in this module, but rather the `read_raster` method should be adapted. Opening an issue on this is certainly a good idea.

Reply:
However, don't you also have some potential `Landslide` specific issues in the part here 

```
        ls_gdf_bbox = ls_gdf_bbox.assign(row=np.nan, col=np.nan)
        ls_gdf_bbox[['col', 'row']] = ls_gdf_bbox.apply(
            lambda row: mapping_point2grid(row.geometry, bbox[-1], bbox[0], 
                                           res), axis = 1).tolist()
        ls_gdf_bbox['flat_ix'] = ls_gdf_bbox.apply(
            lambda row: mapping_grid2flattened(row.col, row.row, 
                                               self.centroids.shape), axis = 1)        
        self.intensity = sparse.csr_matrix(
            (np.ones(n_ev), (np.arange(n_ev), ls_gdf_bbox.flat_ix)), 
            shape=(n_ev, self.centroids.size))
        self.fraction = self.intensity.copy()
        self.frequency = self.intensity.copy()
```
?

Reply:
don't understand the question

Reply:
took this up in issue #172